3D Chess/Assets/Scripts/PieceSpawner.cs

```diff
@@ -18,77 +17,53 @@ public class PieceSpawner : MonoBehaviour
     [SerializeField] private Piece prefabWhiteQueen = default;
     [SerializeField] private Piece prefabWhiteRook = default;
 
-    public void LoadBoard(GameObject parent)
+    public void CreateNewBoard(Realm realm)
```

Comment:
Is there a reason we pass the realm around? Just for simplicity?

Reply:
We already create the instance and instead of calling the constructor again we just pass around the instance that already exists. This should be more efficient than creating it again and again (which won't really be noticeable in this example to be fair).